### ISSUE TITLE
Is there any support for deep update

### ISSUE THREAD

**Issue by itmstr01** [SAP/cloud-sdk-java#630]
I am currently working on a connector to "sap/opu/odata/sap/API_PRODUCT_SRV".
A product has several relations to Description, BasicText ....
I can insert a new Product together with the filled relations and the filled relations are created together with the Product.
On update I have to manually distinct between insert, update and delete for the relation entities!
This is hard work for one relation and leads into multiple code duplication for every relation.
I tried to call the fluent methods using function-references but this does not work because the
function-names contain the entity-names rather using types. (e.g. "...updateProductDescription(entity)")
Please, if there is a solution, give it to the community.

**Comment by newtork** [comment]
I think **deep update** via **PATCH** should work.

But probably not for collection-type navigation-property, is this similar to your described use-case...?
https://github.com/SAP/cloud-sdk-java/blob/main/datamodel/odata-v4/odata-v4-core/src/test/java/com/sap/cloud/sdk/datamodel/odatav4/core/UpdateRequestHelperPatchTest.java#L342-L353

**Comment by itmstr01** [comment]
This V2-code with cloud.sdk 5.13.0
only modifies the Product and not the ProductDescription
```
            try {
                List<ProductDescription> descs = product.getDescriptionOrFetch();
                ProductDescription pd = descs.get(0);
                pd.setProductDescription("My changed Description");
                ModificationResponse<Product> modiResp = new DefaultProductSrvService()
                        .updateProduct(product)
                        .executeRequest(dest);
                statuscode = modiResp.getResponseStatusCode();
                product = modiResp.getModifiedEntity();
            } catch (ODataServiceErrorException e) {
```
To modify the ProductDescription I have to update i separately! (not to speak about create and delete additionally)
And this update-code i have to repeat for every relation (Description, BasicText ...).
I tried to create a reusable update-code but I cannot find a solution.
Here my experience:
```
       DefaultAPIPRODUCTSRVService sv = new DefaultAPIPRODUCTSRVService();
       Function<BasicText, FluentHelperUpdate<?, ?>> updateFunc = sv::updateProductBasicText;
       Function<?, FluentHelperUpdate<?, ?>> ggg = updateFunc;

       Function0<List<?>> getOrFetchFunc = product::getProductBasicTextOrFetch;

       List<VdmEntity<?>> btxs = (List<VdmEntity<?>>) getOrFetchFunc.apply();//product.getProductBasicTextOrFetch();
            for (VdmEntity<?> bt : btxs) {
                  FluentHelperUpdate<?, ?> u = ggg.apply(bt); //sv.updateProductBasicText(bt);
                  ModificationResponse<?> v = u.executeRequest(dest);
                  statuscode = v.getResponseStatusCode();
                  if (statuscode != 204) { // no content (update)
                       throw new Exception...
                   }
             }
```
"ggg.apply(bt);" is not possible because java does not know that updateProductBasicText() schould be used.

**Comment by newtork** [comment]
> only modifies the Product and not the ProductDescription
> _[code]_

Yeah, as I said one-to-many navigation-properties are not supported for deep-update.

* Have you tried using `FluentHelperUpdate#includingFields(...)` to force fields to be considered in update requests?

* Alternatively have you tried switching from PATCH (`FluentHelperUpdate#modifyingEntity()`) to PUT (`FluentHelperUpdate#replacingEntity()`)?
  The former is default behavior; unchanged fields are not being sent to OData service.
  The latter is for replacing the whole entity payload, including collection-type properties.
  
  Please be aware, depending on the service "replacingEntity" may lead to information loss, if not all data was retrieved before. 
  PUT requires all entity field values to be present, `null` values may lead to entity field value deletion.

* Alternatively you could serialize everything to a JSON string and run a low-level OData client request.

* Or you implement your own entity field discovery logic, as mentioned in your last comment. We cannot support/consult though.

**Comment by KavithaSiva** [comment]
@itmstr01 I am closing this issue due to inactivity, please feel free to reopen if you have further questions.
